code/datums/components/archaeology.dm

```diff
@@ -0,0 +1,85 @@
+/* How this works:
+		In turf.Initialize() , AddComponent(/datum/component/archaeology, prob2drop). prob2drop is a base number that affects each drop in list/drop the same. Good for if you want to have turfs be a bit more dynamic.
+		In the turf's vars, you can set the drop list with the archdrops var. Format is type = num where num is the max it could possibly drop. postdig
+		In the turf's AttackBy() [open turfs are vastly different and don't typically call inheritance hence component], call ArchaeologySignal(user, W). This will send the signal to the component as well.
+		If your turf has a unique post-dig sprite like basalt/asteroid, put that in postdig_icon as a text string AND set postdig_icon_change = TRUE.
+*/
+/datum/component/archaeology
+	dupe_type = COMPONENT_DUPE_UNIQUE
+	var/list/drops = list()
+	var/prob2drop
```

Comment:
Declare this as a number like var/prob2drop = 0 (or whatever). If you want to get really fancy, you could do var/prob2drop = 0 in 0 to 100

Reply:
It's already clamped in init